TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/TrafficReplayer.java

```diff
@@ -397,24 +414,31 @@ void runReplayWithIOStreams(Duration observedPacketConnectionTimeout,
         }
     }
 
+    ConcurrentHashMap<UniqueRequestKey, Boolean> liveRequests = new ConcurrentHashMap<>();
+
     private BiConsumer<UniqueRequestKey, HttpMessageAndTimestamp>
     getRecordedRequestReconstructCompleteHandler(ReplayEngine replayEngine) {
         return (requestKey, request) -> {
             replayEngine.setFirstTimestamp(request.getFirstPacketTimestamp());
 
+            liveTrafficStreamLimiter.addWork(1);
```

Comment:
If we have allotted a max of 1024 concurrent requests which are all full, will we block right here until space becomes available?

Reply:
Okay and we have other logic to say we are at max concurrent requests with requests being blocked don't take in anymore Kafka records?

Reply:
addWork (from this callback) is run from within pullCaptureFromSourceToAccumulator, so blocking within the callback means that the entire accumulation thread is blocked.  pullCaptureFromSourceToAccumulator pulls from Kafka or any other producer.  The Kafka traffic source (& others) can only read an bounded number of requests with each call to readNextTrafficStreamChunk().  By blocking here, that puts necessary constraints on when we'll be bringing more captures/requests into the process.